osu.Game/Screens/OnlinePlay/Lounge/Components/RoomPanel.cs

```diff
@@ -556,5 +555,57 @@ private void load(OsuColour colours)
                 };
             }
         }
+
+        public partial class RoomNameLine : FillFlowContainer
+        {
+            private readonly string? roomUrl;
+            private readonly bool showExternalLink;
+
+            private TruncatingSpriteText spriteText = null!;
+            private ExternalLinkButton link = null!;
+
+            public LocalisableString Text
+            {
+                get => spriteText.Text;
+                set => spriteText.Text = value;
+            }
+
+            public RoomNameLine(string? roomUrl, bool showExternalLink)
+            {
+                this.roomUrl = roomUrl;
+                this.showExternalLink = showExternalLink;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load()
+            {
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+                Direction = FillDirection.Horizontal;
+
+                Children = new Drawable[]
+                {
+                    spriteText = new TruncatingSpriteText
+                    {
+                        Anchor = Anchor.BottomLeft,
+                        Origin = Anchor.BottomLeft,
+                        Font = OsuFont.GetFont(size: 28),
+                    },
+                    link = new ExternalLinkButton(roomUrl)
```

Comment:
Not sure it should be fixed here, but this button taking in a potential null url and still displaying as if nothing is wrong is very weird.

Reply:
Yes that's what I'm saying is weird. Check the internal implementation. It still shows the button but using it does nothing. It's not disabled or anything. Does not feel good.

Reply:
Addressed in https://github.com/ppy/osu/pull/33858/commits/8b67747735347a21867ecac7c9e5fe8d7459fb8b by hiding the button if a link is not available.